### PR TITLE
WIP BXMSDOC-6503: Document dashbuilder external components feature.

### PR DESCRIPTION
**Pull request by kaldesai**
**JIRAs:** 
- [Epic](https://issues.redhat.com/browse/BXMSDOC-6276) 
- [Dedicated JIRA](https://issues.redhat.com/browse/BXMSDOC-6503)

**Doc previews:**
- [RHPAM-Building custom dashboard widgets](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-6503-RHPAM-externalcomponents/#building-custom-dashboard-widgets-external-components-con)
- [RHDM-Building custom dashboard widgets](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-6503-RHDM-externalcomponents/#building-custom-dashboard-widgets-external-components-con)

**Doc impact:** 
- For **RHPAM**, go to IV. Building custom dashboard widgets ---> Chapter **57.5. External components**.
- For **RHDM**, go to IV. Building custom dashboard widgets  ---> Chapter **51.5. External components**.

### CODE REVIEW

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
+[id='building-custom-dashboard-widgets-external-components-con']
+= External components
+
+In {CENTRAL}, a page can have external components. The components are disabled by default. To enable the external components, change the value of `dashbuilder.components.enable` system property to `true`.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc:4]:
```suggestion
In {CENTRAL}, you can add external components to a page. The components are disabled by default. To enable the external components, change the value of `dashbuilder.components.enable` system property to `true`.
```

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+The external component location is set and configured with the `dashbuilder.components.dir` system property. The default value of this system property is `/tmp/dashbuilder/components`.
+You must set the component under components directory with a parent directory, that will be the component's ID. For example, if the component ID is `mycomp` and the component directory is `/tmp/dashbuilder/components`, then the component base directory is `/tmp/dashbuilder/components/mycomp`.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc:7]:
```suggestion
You must set the component under the components directory with a parent directory, which is used as the component ID. For example, if the component ID is `mycomp` and the component directory is `/tmp/dashbuilder/components`, then the component base directory is `/tmp/dashbuilder/components/mycomp`.
```

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+|Name of the component displayed under *Components* section.
+
+|`icon`
+|Icon of the component displayed under *Components* section.
+
+|`noData`
+|A flag that indicates that the component does not require a data set.
+
+|`parameters`
+|The list of parameters supported by the component uses `ComponentParameter` type. Supported parameter types are `name`, `type`, `category`, `defaultValue`, `label`, and `comboValues`.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc:27]:
```suggestion
|The list of parameters using `ComponentParameter` type. Supported parameter types include `name`, `type`, `category`, `defaultValue`, `label`, and `comboValues`.
```

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
+[id='building-custom-dashboard-widgets-creating-external-components-proc']
+= Creating external components
+
+The following procedure describes how to create and add the external components:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc:4]:
```suggestion
The following procedure describes how to create and add external components to a page:
```

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+.Procedure
+
+. Set the component under components directory with a parent directory.
++
+For example, if the component ID is `mycomp` and the component directory is `/tmp/dashbuilder/components`, then the component base directory is `/tmp/dashbuilder/components/mycomp`.
+. Create the `manifest.json` file in the component directory.
+. Create `index.html` file with any HTML content.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc:12]:
```suggestion
. Create `index.html` file with HTML content.
```

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+[source]
+----
+$ ~/EAP_HOME/bin/standalone.sh -c standalone-full.xml
+-Ddashbuilder.components.dir={component directory base path} -Ddashbuilder.components.enable=true
+----
+. Start {CENTRAL}, go to *Menu → Design → Pages*.
++
+*External Components* is available under *Components* pane.
+
+. In the *Components* pane, expand the *External Components* and drag the required component types to the *Editor* canvas.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc:25]: Do we need to bold editor? 
```suggestion
. In the *Components* pane, expand the *External Components* and drag the required component types to the canvas.
```

**Review by hmanwani-rh — COMMENTED** [review]
@kaldesai Some minor suggestions added below. Rest looks good to me. 
Please let me know if you have any questions.

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc (hunk 1 above)
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc:4]: Done.

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc (hunk 2 above)
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc:7]: Done.

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc (hunk 3 above)
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc:27]: Done

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc (hunk 4 above)
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc:4]: Done

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc (hunk 5 above)
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc:12]: Done

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc (hunk 6 above)
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-external-components-proc.adoc:25]: Done

**Pull request merged**